src/features/packages/packageRepository.ts

```diff
@@ -56,7 +56,18 @@ export class PackageRepository {
             tasks.push(this.packageUpload(spaceId, packagePath, overwriteMode));
         }
 
-        await Promise.allSettled(tasks);
+        const rejectedTasks: unknown[] = [];
+
+        const completedTasks = await Promise.allSettled(tasks);
+        for (const t of completedTasks) {
+            if (t.status === "rejected") {
+                rejectedTasks.push(t.status);
+            }
+        }
+
+        if (rejectedTasks.length > 0) {
+            throw new Error(`${rejectedTasks}`);

```

Comment:
This should be changed to something more useful, like:
```
const errors: Error[] = [];
for (const e of rejectedTasks) {
    if (e instanceof Error) {
        errors.push(e)
    } else {
        errors.push(new Error(`unexpected error: ${e}`))
    }
}
throw new AggregateError(errors, "failed to upload all packages")
```
This way we don't lose any of the information from any of the errors that happen along the way.

Reply:
fixed